### DISCUSSION TITLE
Broken link in 'authenticator replaced' email

### DISCUSSION THREAD

**Discussion by pettinen** [npm/feedback#684] [Website] [answered]
The email titled *[npm] Authenticator was replaced on your account* has a broken 'view your recovery codes' link, because the URL is surrounded in backticks: ``<a href=`https://www.npmjs.com/settings/pettinen/recovery-codes`>``

**Comment by MylesBorins** [comment, marked as answer]
I was not able to reproduce this.

can you please [open a ticket](https://www.npmjs.com/support) with support about this including the time you received the email, the email address your received it at, and the related account.

**Reply by pettinen** [reply]
Submitted.

**Reply by nehsharmMS** [reply]
@pettinen I am unable to reproduce the issue. The link in <a> tag seems fine with double quotes. 

<img width="715" alt="recovery-codes-email" src="https://user-images.githubusercontent.com/60141664/170208211-8935368f-3845-4f8d-a7a2-5063523b0a06.png">

Can you retry and see if the issue still persist? If yes then please provide more information i.e. which browser and mail client you are using.

**Reply by pettinen** [reply]
I'm using the ProtonMail webmail with Chrome 102.0.5005.63.

Gmail, which I also tried with just now, seems to be smart enough to fix the link, but even in Gmail the backticks are present in the original message HTML:
```
Please <a href=3D`https://www.npmjs.com/settings/pettinen/recovery-codes`>v=
iew your recovery codes</a> and save them somewhere safe and handy. Recover=
y codes are the only way to access your account again.<br/>=20
```

**Reply by nehsharmMS** [reply]
The above screenshot is from Gmail html. I tried with outlook and ProtoMail webmail too. It still works for me. I am using chrome 102.0.5005.61

<img width="624" alt="Screenshot 2022-05-25 at 4 50 47 PM" src="https://user-images.githubusercontent.com/60141664/170250969-01b375cf-2018-4e42-9685-91fa24caef4a.png">

Edit : added the screenshot

**Reply by pettinen** [reply]
Like I said, Gmail seems to fix the link for display so it doesn't show up in dev tools. However, the invalid HTML is there when I check the raw message ("Show original" in Gmail).

Tested again, still broken in ProtonMail in Chrome 102.0.5005.63 and their mobile app.
![image](https://user-images.githubusercontent.com/18386355/170249833-3275e082-b9d3-470f-ab4b-5996461ea12d.png)

**Reply by nehsharmMS** [reply]
@pettinen RAW html is also valid for me - gmail client chrome 102.0.5005.61. Same true for ProntonMail client. 

<img width="660" alt="Screenshot 2022-05-25 at 5 11 38 PM" src="https://user-images.githubusercontent.com/60141664/170253975-742baabd-54e4-4121-89f1-c1fa113899d7.png">

Which OS are you using?

**Reply by pettinen** [reply]
How strange. OS:
Edition	Windows 10 Enterprise
Version	21H2
OS build	19044.1706
Experience	Windows Feature Experience Pack 120.2212.4170.0

**Reply by nehsharmMS** [reply]
Thanks @pettinen. Can you also check if `preferences` in the email footer works for you?

**Reply by pettinen** [reply]
Yes, all other links in the email are as they should.

**Reply by nehsharmMS** [reply]
Thanks @pettinen. We are investigating the issue.

**Reply by arjgupta** [reply]
@pettinen We have made a change to fix this issue. The link to the recovery codes page should be enclosed in double quotes now. Let us know if you're still facing this issue

**Reply by pettinen** [reply]
I will confirm that the issue seems to be fixed now. Sorry for the delay.
